spinn_front_end_common/interface/provenance/provides_provenance_data_from_machine_impl.py

```diff
@@ -210,64 +211,65 @@ def _read_basic_provenance_items(self, provenance_data, placement):
-                label, x, y, p, callback_queue_overloaded)))
-
-        data_items.append(ProvenanceDataItem(
-            self._add_name(names, self._TIMES_DMA_QUEUE_OVERLOADED),
-            dma_queue_overloaded,
-            report=dma_queue_overloaded != 0,
-            message=self._TIMES_DMA_QUEUE_OVERLOADED_MESSAGE.format(
-                label, x, y, p, dma_queue_overloaded)))
-
-        data_items.append(ProvenanceDataItem(
-            self._add_name(names, self._TIMER_TICK_OVERRUN),
-            number_of_times_timer_tic_over_ran,
-            report=number_of_times_timer_tic_over_ran != 0,
-            message=self._TIMER_TICK_OVERRUN_MESSAGE.format(
-                label, x, y, p, number_of_times_timer_tic_over_ran)))
-
-        data_items.append(ProvenanceDataItem(
-            self._add_name(names, self._MAX_TIMER_TICK_OVERRUN),
-            max_number_of_times_timer_tic_over_ran,
-            report=max_number_of_times_timer_tic_over_ran > 0,
-            message=self._MAX_TIMER_TICK_OVERRUN_MESSAGE.format(
-                label, x, y, p, max_number_of_times_timer_tic_over_ran)))
-
-        return data_items
+        return [
+            ProvenanceDataItem(
+                names + [self._TIMES_TRANSMISSION_SPIKES_OVERRAN], tx_overflow,
+                report=(tx_overflow != 0),
+                message=self._TIMES_TRANSMISSION_SPIKES_OVERRAN_MESSAGE.format(
+                    label, x, y, p, tx_overflow)),
+            ProvenanceDataItem(
+                names + [self._TIMES_CALLBACK_QUEUE_OVERLOADED],
+                callback_overload,
+                report=(callback_overload != 0),
+                message=self._TIMES_CALLBACK_QUEUE_OVERLOADED_MESSAGE.format(
+                    label, x, y, p, callback_overload)),
+            ProvenanceDataItem(
+                names + [self._TIMES_DMA_QUEUE_OVERLOADED], dma_overload,
+                report=(dma_overload != 0),
+                message=self._TIMES_DMA_QUEUE_OVERLOADED_MESSAGE.format(
+                    label, x, y, p, dma_overload)),
+            ProvenanceDataItem(
+                names + [self._TIMER_TICK_OVERRUN], tic_overrun_count,
+                report=(tic_overrun_count != 0),
+                message=self._TIMER_TICK_OVERRUN_MESSAGE.format(
+                    label, x, y, p, tic_overrun_count)),
+            ProvenanceDataItem(
+                names + [self._MAX_TIMER_TICK_OVERRUN], tic_overrun_max,
+                report=(tic_overrun_max > 0),
+                message=self._MAX_TIMER_TICK_OVERRUN_MESSAGE.format(
+                    label, x, y, p, tic_overrun_max))]
 
     def _get_remaining_provenance_data_items(self, provenance_data):
         """
-        :param list(ProvenanceDataItem) provenance_data:
-        :rtype: list(ProvenanceDataItem)
+        :param list(int) provenance_data:
+        :rtype: list(int)
         """
         return provenance_data[self.NUM_PROVENANCE_DATA_ENTRIES:]
 
+    def _get_extra_provenance_items(
```

Comment:
I do not like a function to be overwitten to start with an underscore.

Reply:
Why is location a tuple when the only place calling it has x, y, p and has to make a tuple and all places that use it split it back out to x, y, p

Reply:
They only split it out so they can use it (in inconsistent ways!) in messages that are only parsed by humans, not machines.

Reply:
Method is now called `parse_extra_provenance_items`.